### PR TITLE
Add support for fine-tuning MMS mdoel via adapters

### PR DESCRIPTION
**Pull request by Lguyogiro**
This PR adds a `finetune_mms_adapter.py` script that follows the whisper fine-tuning script, but uses the 1 billion parameter Massively Multilingual Speech model from facebook. The fine tuning is done by freezing the base model and adding/updating adapter layers.  I've tested it on a number of languages and it works.

Some consideration:
- this is a useful workflow as the results can sometimes be better than Whisper, particularly for languages not covered by Whisper's pretraining (the MMS pretraining included a lot of languages, and this script adds a new monolingual processor for fine-tuning so you can get better results for unseen languages). However, the adapter-layer fine tuning, as implemented, ONLY works for the `facebook/mms-1b-all` model. The smaller model in hf does not support adapters. Thus, we should probably add standard wav2vec2 fine-tuning (without adapters) to this repo, which could also be used on the smaller mms model.
- I added an iso-639 code lookup so that the model card can use the code for any valid language that is included in the config.
- The default hyperparameters for Whisper do not work out-of-the-box for this approach. Specifically, you should increase the learning rate for mms+adapter fine-tuning (1e-3 instead of 1e-5).

### CODE REVIEW

**Review by Kostis-S-Z — COMMENTED** [review]
Looks awesome, thank you for following the same structure as `finetune_whisper` so closely, made it super easy to review! I managed to finetune mms-1b on my GPU on the Greek CV for 50 steps with a batch size of 2 :sweat_smile: 

One small thing is that I did see your comment:
>  Since the MMS workflow doesn't require a ton of preprocessing, we don't worry about saving the "processed" dataset (hence the _ for the processed_dataset_path). We do need to make sure the sampling rate is 16k though

But I am not sure if I agree 100%. I tested with the Greek CV (Scripted) that is ~750mb and it took 6-7 minutes to process the train & test subsets every time I run the script. I could see how for larger datasets it could be much longer. 

I think its totally fine to merge the PR as is, but if you have time to add the option to save the processed dataset & load it if it exists, it would be a welcome addition. But again, non blocking to merge the PR!

Finally, I pushed some tiiiiny changes:
- Separate the `config.yaml` to `config_whisper.yaml` and `config_mms.yaml` 
- Added a tiny e2e test for MMS (but excluding it from the GitHub CI workflow since its a bit too heavy)
- Formatting using the `pre-commit`

**Comment by Kostis-S-Z** [comment]
https://github.com/Mozilla-Data-Collective/speech-to-text-finetune/issues/16
